piet-direct2d/src/lib.rs

```diff
@@ -430,17 +439,22 @@ impl<'a> D2DRenderContext<'a> {
 
         // TODO: do something special (or nothing at all) for line?
         if let Some(rect) = shape.as_rect() {
-            self.rt.fill_rect(rect, &brush)
+            self.rt.fill_rect(rect, &brush);
+            return;
         } else if let Some(round_rect) = shape.as_rounded_rect() {
-            self.rt.fill_rounded_rect(round_rect, &brush)
-        } else if let Some(circle) = shape.as_circle() {
-            self.rt.fill_circle(circle, &brush)
-        } else {
-            match path_from_shape(self.factory, true, shape, fill_rule) {
-                Ok(geom) => self.rt.fill_geometry(&geom, &brush, None),
-                Err(e) => self.err = Err(e),
+            if let Some(radius) = round_rect.radii().as_single_radius() {
```

Comment:
I think we can keep this as before, just changing,

```rust
if let Some(round_rect) = shape.as_rounded_rect().filter(|r| r.as_single_radius().is_some()) {
}
```

This isn't a major point, but it cleans up the diff somewhat.